components/plan-build/bin/hab-plan-build.sh

```diff
@@ -177,6 +177,18 @@
 # pkg_expose=(80 443)
 # ```
 #
+# ### pkg_exports
+# An associative array representing configuration data which should be gossiped to peers. The keys
+# in this array represent the name the value will be assigned and the values represent the toml path
+# to read the value.
+# ```
+# pkg_exports=(
+#   [key]=token
```

Comment:
Good question! I totally forgot to follow up on this with you after our discussion before I got in and implemented it.

My thinking was that we could restructure the keyspace of the bind to look more similar to that of the `ServiceConfig`, but with a smaller surface. In this PR I began to flatten the namespace by putting all of the exported configuration in `bind.{member}.cfg`. I was thinking that we'd also probably want to gossip the contents of our `ServiceConfig.sys` as well and that'd become `bind.{member}.sys`. I don't think we need to actually perform the mapping here because the contents of `ServiceConfig.sys` is both small and already structured for us. Every peer will have the same sys.

I can see us also wanting to export a small portion of the `Pkg` struct as `bind.{member}.pkg` as well. I don't think we need to share all of it - for example, the consumer doesn't care about the producer's exports, path, or really any of the in the svc_* fields. They're more of a private member and only usable by the producer's side.

After writing this out to you I think I'm going to go ahead and make these changes in this PR and we should continue the discussion about it before merging!